src/App.jsx

```diff
@@ -14,33 +16,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ABC = 'abc';
+const SORT_BY_LENGTH = 'leng';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getGoodsView(sortField, reverse) {
+  let prepearedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (sortField) {
+    case SORT_ABC:
+      prepearedGoods = prepearedGoods.sort((cur, prew) => {
+        return reverse ? prew.localeCompare(cur) : cur.localeCompare(prew);
+      });
+      break;
+    case SORT_BY_LENGTH:
+      prepearedGoods = reverse
+        ? prepearedGoods.sort((a, b) => a.length - b.length).reverse()
+        : prepearedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      prepearedGoods = reverse
+        ? prepearedGoods.reverse()
+        : [...goodsFromServer];
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverce] = useState(false);
```

Comment:
There is a typo in the state variable name `setReverce`. It should be `setReverse` to match the intended functionality.